add.go

```diff
@@ -345,6 +345,17 @@ func (b *Builder) addHelper(excludes *fileutils.PatternMatcher, extract bool, de
 
 			b.ContentDigester.Start("file")
 
+			if excludes != nil {
```

Comment:
Can we add a comment here explaining what we're doing?

Reply:
fixed